app/components/file-viewer.jsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import VideoPlayer from './video-player';
+import TextViewer from './text-viewer';
+import ImageViewer from './image-viewer';
+
+const VIEWERS = {
+  image: ImageViewer,
+  text: TextViewer,
+  video: VideoPlayer
+};
+
+function FileViewer(props) {
+  const Viewer = VIEWERS[props.type];
+  return (
+    <Viewer
+      src={props.src}
+      type={props.type}
```

Comment:
`type` is probably redundant now. Need to check if any of the specific viewer components are using it.